### PR TITLE
Log model source and class name in the `sentence-transformers` flavor

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/11529?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11529/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11529
```

</p>
</details>

### What changes are proposed in this pull request?

Log `source_model_name` and `pipeline_model_type` fields in the `sentence-transformers` flavor, similar to the `transformers` flavor. These fields are required for computing provisioned throughput in Databricks model serving.

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by WeichenXu123 — APPROVED** [review]
LGTM!

File: tests/sentence_transformers/test_sentence_transformers_model_export.py [hunk 1]
```diff
@@ -91,11 +93,21 @@ def test_logged_data_structure(model_path, basic_model):
...
+    pyfunc_flavor = mlmodel["flavors"]["python_function"]
+    assert pyfunc_flavor["loader_module"] == "mlflow.sentence_transformers"
+    assert pyfunc_flavor["data"] == mlflow.sentence_transformers.SENTENCE_TRANSFORMERS_DATA_PATH
+
+    st_flavor = mlmodel["flavors"]["sentence_transformers"]
+    assert st_flavor["pipeline_model_type"] == "BertModel"
+    if Version(sentence_transformers.__version__) >= Version("2.3.0"):
+        assert st_flavor["source_model_name"] == "sentence-transformers/all-MiniLM-L6-v2"
+    else:
+        # Before Transformers 2.3.0, the library loads the Transformers model after local snapshot
```
Comment (BenWilson2) [inline tests/sentence_transformers/test_sentence_transformers_model_export.py:107]: Is this behavior useful to the inference team? 
With all current DBR having sentence-transformers 2.2.2 installed, should we handle this by stripping the local file portion out?

Comment (BenWilson2) [reply tests/sentence_transformers/test_sentence_transformers_model_export.py:107]: Per the feedback from Maddie, let's strip out the cache path information and parse the constructed cache name
`/some/local/cache/path/location/sentence_transformers/<hf name>_<model_name>` -> `<hf_name>/<model_name>`

File: tests/sentence_transformers/test_sentence_transformers_model_export.py [hunk 2]
```diff
@@ -91,11 +91,36 @@ def test_logged_data_structure(model_path, basic_model):
...
+
+@pytest.mark.parametrize(
+    ("model_name", "expected"),
+    [
+        (
+            "sentence-transformers/all-MiniLM-L6-v2",
+            "sentence-transformers/all-MiniLM-L6-v2",
+        ),
+        (
+            "/path./to_/local-/path?/sentence-transformers_all-MiniLM-L6-v2/",
```
Comment (BenWilson2) [inline tests/sentence_transformers/test_sentence_transformers_model_export.py:113]: haha nice horror-show path

**Review by BenWilson2 — APPROVED** [review]
Excellent! LGTM!

**Pull request merged**